datafusion/execution/src/config.rs

```diff
@@ -140,6 +140,11 @@ impl SessionConfig {
         &self.options
     }
 
+    /// Returns the config options as an Arc
+    pub fn options_arc(&self) -> Arc<ConfigOptions> {
+        Arc::clone(&self.options)
```

Comment:
I think we should change this to just be an &Arc

It might also be worth changing `options()` directly to return `&Arc` 🤔 

```suggestion
    /// Returns the config options as an Arc
    pub fn options_arc(&self) -> &Arc<ConfigOptions> {
        &self.options
```